src/storage/src/source/reclock.rs

```diff
@@ -185,29 +186,64 @@ where
         // We need to compute the answer by iterating over the consolidated remap trace which will
         // present to us one diff at a time. We know that by construction at any given IntoTime
         // time the remap collection accumulates to a well formed antichain. That is, it contains
-        // exactly one copy of mutually incomparable FromTime elements. Therefore, when we see a
-        // positive diff of some element `from_t` occuring at some timestamp `into_t` we also know
-        // that the FromTime frontier at `into_t` included that element.
+        // exactly one copy of mutually incomparable FromTime elements.
         //
-        // What we do then is that for every positive diff at IntoTime `ts` we check whether the
-        // FromTime element `frontier_ts` is less than or equal to the timestamp to be reclocked
-        // `src_ts`:
-        // * If it is, then the `FromTime` frontier at `ts` *must* be less than or equal to
-        //   `src_ts` and so we know that `src_ts` should *not* be reclocked at `ts` because it is
-        //   beyond the frontier. We codify that by adding `ts` into a mutable antichain with the
-        //   minimum frequency.
-        // * If it is not, then the `FromTime` frontier at `ts` *may* be less than or equal to
-        //   `src_ts` so `src_ts` *may* be reclocked to `ts`. We codify that by adding `ts` into a
-        //   mutable antichain with a frequency of 1.
+        // We also know that `src_ts` is beyond the since frontier, therefore there exist witness
+        // timestamps `from_ts` that are less than or equal to `src_ts` and occur at IntoTime times
+        // with a positive diff. We also know that `src_ts` is not beyond the upper frontier,
+        // therefore all the positive diffs of the witness times must be retracted at subsequent
```

Comment:
what if the trace is compacted to a single value, which is `==` to `src_ts`? Doesn't that mean no witness time exists? we have a positive diff for `src_ts` but no retractions at all?

Reply:
The function checks that `src_ts` is beyond the since and not beyond the upper before continuing which means that the trace must have at least two frontiers at two distinct timestamps so we can't ever have this situtation